pkg/ocm/subscription/service.go

```diff
@@ -49,20 +53,64 @@ type SubscriptionGetter interface {
 func (client *subscriptionClient) GetSubscriptions() ([]Subscription, error) {
 	discovered := []Subscription{}
 	request := client.Config
-
 	request.Page = 1
+
+	logf.V(1).Info("Starting subscription retrieval", "BaseURL", client.Config.BaseURL, "Size", client.Config.Size)
+
 	for {
+		// Logging request details
+		logf.V(2).Info("Sending subscription request", "Page", request.Page, "Size", request.Size)
+
+		// Fetch the subscriptions
 		discoveredList, err := SubscriptionProvider.GetSubscriptions(request)
+
 		if err != nil {
-			return nil, fmt.Errorf(err.Error.Error())
+			if err.Error == nil && err.Reason != "" {
+				err.Error = fmt.Errorf(err.Reason)
```

Comment:
Optional: Errorf will be processing the Reason as a format string.  If it has a format character in it another parameter would be expected.

Reply:
Updated this line to use `err.Error = errors.New(err.Reason)`.